cmd/generate/config/rules/telegram.go

```diff
@@ -0,0 +1,57 @@
+package rules
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func TelegramBotToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Telegram Bot API Token",
+		RuleID:      "telegram-bot-api-token",
+		SecretGroup: 1,
+		Regex:       regexp.MustCompile(`(?i)(?:^|[^0-9])([0-9]{5,16}:A[a-zA-Z0-9_\-]{34})(?:$|[^a-zA-Z0-9_\-])`),
```

Comment:
@b4bay 
I'm concerned that this doesn't follow the same patterns as the other rules. Even though you have `telegram` in the keywords, this rule could still be incorrectly associating matches since you have other strings like `api, bot, token, url`, all of which are generic. Is `:A` suppose to be the uniquely identifiable string in a telegram token?

Reply:
@zricethezav 

Your concerns are understandable.
`regexp.MustCompile` is used instead of recommended `generateSemiGenericRegex` / `generateUniqueTokenRegex` just because there is no way to match all the cases mentioned in https://github.com/zricethezav/gitleaks/pull/121#issuecomment-435955348. Or at least I wasn't lucky enough to find it.

> Is :A suppose to be the uniquely identifiable string in a telegram token?

Of course not. I bet that the coincidence of all four conditions:

1. 5-16 digits, following by
2. Exactly 35 alphanumerics
3. Starting with `A`
4. And have `telegram`, `api`, `bot`, `token` or `url` in the variable name

is unlikely.

Reply:
@b4bay that's fair. If it turns out this is generating too many false positives we can always tune it more. If you resolve the conflicts I'll go ahead and merge

Reply:
@zricethezav resolved by 9f2379a5639483fce1a95caf9e78743e1ff3db7a